### PR TITLE
fix(Button): remove button as default role

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Remove unnecessary button default role 

Pointed out [here](https://codecademy.atlassian.net/browse/GM-576)

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [ABC-123]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Checkout the button storybook
2. They should still look like buttons and not have an additional "button" role

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monorepo    | [Monorepo PR](https://github.com/codecademy-engineering/mono/pull/6252)  | [Monolith Env](http://www.google.fr/ 'Named link title') |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.19.1-alpha.8d02f2.0<br/>@codecademy/gamut-kit@0.6.407-alpha.8d02f2.0<br/>@codecademy/styleguide@66.19.3-alpha.8d02f2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://663e8f6b329d2a57dace1cd7--gamut-preview.netlify.app>https://663e8f6b329d2a57dace1cd7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/663e8f6b329d2a57dace1cd7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.19.1-alpha.742607.0<br/>@codecademy/gamut-kit@0.6.407-alpha.742607.0<br/>@codecademy/styleguide@66.19.3-alpha.742607.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://664266ff7db7c61678ad92e1--gamut-preview.netlify.app>https://664266ff7db7c61678ad92e1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/664266ff7db7c61678ad92e1>Deploy Logs</a></p>

File: packages/gamut/src/ButtonBase/__tests__/ButtonBase.test.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  it('renders a button with the correct role', () => {
+    const { view } = renderView({ role: 'menuitem' });
+
+    view.getByText(buttonText);
+    view.getByRole('menuitem');
+  });
+
+  describe('when the button is disabled', () => {
+    it('does not render the href attribute and renders a button', () => {
+      const buttonText = 'Submit';
```
Comment (LinKCoding) [inline packages/gamut/src/ButtonBase/__tests__/ButtonBase.test.tsx:42]: does this need to be set again? doesn't it have access to `buttonText` in line 5?

File: packages/gamut/src/ButtonBase/__tests__/ButtonBase.test.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      const { view } = renderView({ disabled: true });
+
+      const el = view.getByText(buttonText);
+      view.getByRole('button');
+      expect(el.getAttribute('href')).toBeNull();
+    });
+
+    it('renders the disabled attribute on button tags', () => {
+      const buttonText = 'Submit';
```
Comment (LinKCoding) [inline packages/gamut/src/ButtonBase/__tests__/ButtonBase.test.tsx:51]: same Q here about setting `buttonText`

**Review by LinKCoding — APPROVED** [review]
lgtm! no longer seeing `role="button"` for the buttons rendered in SB :) 
I had some questions about the test, but more so to learn and not blockers.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.19.1-alpha.946e84.0<br/>@codecademy/gamut-kit@0.6.407-alpha.946e84.0<br/>@codecademy/styleguide@66.19.3-alpha.946e84.0<br/></pre>

**Pull request merged**